tests/python/rest_api/_test_base.py

```diff
@@ -522,8 +525,25 @@ def _uploaded_video_task_fxt_base(
             range(start_frame or 0, (stop_frame or frame_count - 1) + 1, step or 1)
         )
 
-        video_file = generate_video_file(frame_count)
-        video_data = video_file.getvalue()
+        assert (not video_file) ^ (
+            chapters is not None
+        ), "Chapters must be specified with a custom video file"
```

Comment:
Does this condition really need to be here? It seems like these parameters could be processed independently:

```
if chapters is None:
    chapters = [...]

if video_file is None:
    video_file = ...
```